### PR TITLE
Null Safety

### LINKED ISSUE #548: Null Safety
Are there plans to add null safety support, and if so: when do you think this might happen.

Thanks!

### PR DESCRIPTION
**Pull request by jlubeck**
This is an attempt at supporting null safety on this awesome library.

All current tests are passing without any changes to them, which is a good sign.

Fixes #548

### CODE REVIEW

**Comment by jlubeck** [comment]
Not sure how to make the tests here run on dart >=2.12 (that's what's failing here, running the tests locally give no errors)

File: example/pubspec.yaml [hunk 1]
```diff
@@ -17,17 +17,12 @@ environment:
...
-  flutter:
-    sdk: flutter
-
+  cupertino_icons: ^0.1.2
   fl_chart:
     path: ../
-
-  # The following adds the Cupertino Icons font to your application.
-  # Use with the CupertinoIcons class for iOS style icons.
-  cupertino_icons: ^0.1.2
```
Comment (imaNNeoFighT) [inline example/pubspec.yaml:28]: Why you removed it?

**Comment by imaNNeoFighT** [comment]
Did you do that using `dart migrate` package?
Because I think we must do more investigation to get it done.
As it is important a lot to people. I will migrate within two days.

File: example/pubspec.yaml (hunk 1 above)
Comment (jlubeck) [reply example/pubspec.yaml:28]: I didn't remove it, I put it on top because your linter was complaining that the dependencies were not sorted

**Comment by jlubeck** [comment]
Hey @imaNNeoFighT I started with the `dart migrate` but then went through it manually as well. 
I'm using this release now on my project (since it required the null safety) and I haven't seen any problems yet. Plus like I mentioned all the tests pass without modification.

But if you want to do it yourself, by all means go for it! Was just trying to help

**Comment by imaNNeoFighT** [comment]
I appreciate your great help. It is a lot for me.
As I have a deep understanding of the source code I will do that with all respect.
Thanks for helping!

**Comment by imaNNeoFighT** [comment]
Check [this branch](https://github.com/imaNNeoFighT/fl_chart/tree/0.20.0-nullsafety1) please.

**Pull request closed without merging**